### ISSUE TITLE
Parsing error: Klaxon drops parameters when using @TypeFor

### ISSUE THREAD

**Issue by wolpl** [cbeust/klaxon#243]
I try to deserialize objects of classes that are derieved from the `Sensor` class:
```kotlin
@TypeFor(field = "type", adapter = SensorTypeAdapter::class)
abstract class Sensor(val type: Type) {
    enum class Type { T1, T2, Test }
}
```
A test subclass looks like this:
```kotlin
class Test(val prop: String) : Sensor(Type.Test) 
```
And the TypeAdapter like this:
```kotlin
import com.beust.klaxon.TypeAdapter
import kotlin.reflect.KClass

class SensorTypeAdapter : TypeAdapter<Sensor> {
    override fun classFor(type: Any): KClass<out Sensor> =
        when (Sensor.Type.valueOf(type as String)) {
            Sensor.Type.Test -> Test::class
            else -> throw IllegalArgumentException("Unknown sensor type $type")
        }
}
```

When trying to parse this I get the exception `com.beust.klaxon.KlaxonException: Unable to instantiate Test with parameters []`. So it seems to me that Klaxon found out which type to use (`Test`) but ignores all other parameters that can be found in the json:
```json
{
    "type": "Test",
    "prop": "String"
}
```

I start the parsing by using a File object:
```kotlin
Klaxon().parse<Sensor>(file)
```

When removing the `prop` property from the json and the `Test` class the deserialisation works.
How do I get it work with properties in my derieved class?

**Comment by cbeust** [comment]
I just created the following test:

```
@Test
class IssuesTest {
    class SensorTypeAdapter : TypeAdapter<Sensor> {
        override fun classFor(type: Any): KClass<out Sensor> =
                when (Sensor.Type.valueOf(type as String)) {
                    Sensor.Type.Test -> Test::class
                    else -> throw IllegalArgumentException("Unknown sensor type $type")
                }
    }

    class Test(val prop: String) : Sensor(Type.Test)

    @TypeFor(field = "type", adapter = SensorTypeAdapter::class)
    abstract class Sensor(val type: Type) {
        enum class Type { T1, T2, Test }
    }


    fun issue243() {
        val json = """
            {
                "type": "Test",
                "prop": "String"
            }
                    """.trimIndent()
        val result = Klaxon().parse<Sensor>(json)
        println(result)
    }
}
```

and it passes, with the correct value in `result`:

![image](https://user-images.githubusercontent.com/92322/52499419-3e2e0880-2b90-11e9-8a3e-e34f0c1d084d.png)

Are you using the latest Klaxon version?

**Comment by wolpl** [comment]
When copying your code into a test (only adding `"""` to end the raw string and annotating the function to make it a junit test) I still get the same error. I am using Klaxon 5.0.1 and Kotlin 1.3.10. Also tried kotlin 1.3.11.

**Comment by wolpl** [comment]
I just created a minimal example project at [https://github.com/wolpl/KlaxonIssue243](https://github.com/wolpl/KlaxonIssue243) where the issue still occurs. this project is using kotlin 1.3.21.

**Comment by cbeust** [comment]
I ran your project (after adding a `data` to your `Test` class for better printing), and it worked right away:

```
Test(prop=String)
Disconnected from the target VM, address: '127.0.0.1:63038', transport: 'socket'
```

Are you using the latest Klaxon?

**Comment by wolpl** [comment]
As the `build.gradle` file shows and I mentioned above I used Klaxon 5.0.1. **_The error disappears when using Klaxon 5.0.5_**. Still it should not have worked in the example project because version 5.0.1 is used there...

@cbeust Thanks for your support. It would have helped to reference the _latest_ version in the project readme - I had to find it manually at bintray.com. As of now applying the instructions in the readme (including version 5.0.1 via gradle) makes it impossible to use the polymorphism features described in the readme.

Perhaps a badge is a convenient solution to always show the latest available version:
 [ ![Download](https://api.bintray.com/packages/cbeust/maven/klaxon/images/download.svg) ](https://bintray.com/cbeust/maven/klaxon/_latestVersion) 
```
 [ ![Download](https://api.bintray.com/packages/cbeust/maven/klaxon/images/download.svg) ](https://bintray.com/cbeust/maven/klaxon/_latestVersion)
```
